data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/consumer/OrderedConsumerVerticle.java

```diff
@@ -223,4 +245,42 @@ private boolean areAllExecutorsBusy() {
     // All executors are busy
     return true;
   }
+
+  private void recordExecutorQueueLatency(final ConsumerRecordContext recordContext) {
+    final var latency = recordContext.performLatency();
+    logger.debug("Executor queue latency {}", keyValue("latency", latency));
+
+    if (meterRegistry != null) {
+      Metrics
+      .executorQueueLatency(getTags(recordContext))
+      .register(meterRegistry)
+      .record(latency);
+    }
+  }
+
+  private void setValueQueueLength(final TopicPartition topicPartition, Queue<Supplier<Future<?>>> executor) {
+    if (meterRegistry != null) {
+      Metrics
+        .queueLength(getTags(topicPartition), executor)
+        .register(meterRegistry)
+        .value();
+    }
+  }
+
+  private Tags getTags(ConsumerRecordContext recordContext) {
+    if (recordContext.getRecord().record().value() instanceof InvalidCloudEvent) {
+      return Tags.of(
+        Tag.of(Metrics.Tags.EVENT_TYPE, "InvalidCloudEvent")
+      );
+    }
+    return Tags.of(
+      Tag.of(Metrics.Tags.EVENT_TYPE, recordContext.getRecord().value().getType())
+    );
+  }
+
+  private Tags getTags(final TopicPartition topicPartition) {
+    return Tags.of(
+      Tag.of(Metrics.Tags.PARTITION_ID, "" + topicPartition.getPartition())
```

Comment:
we probably want to add the topic name as well since a partition id isn't unique.

I'd also add the Kafka resource reference here as well with:

```java
      Tag.of(Metrics.Tags.RESOURCE_NAMESPACE, egress.getReference().getNamespace()),
      Tag.of(Metrics.Tags.CONSUMER_NAME, egress.getReference().getName())
```